src/components/results/AutocalsolResultGraph.vue

```diff
@@ -0,0 +1,169 @@
+<script setup lang="ts">
+import { computed } from 'vue'
+import {
+  convertDataForGraph,
+  calculateIntersectionData,
+  generateXAxis,
+  generateTickPositions,
+} from '@/services/graphService'
+
+import sun from '@/assets/icons/sun.svg'
+import moon from '@/assets/icons/moon.svg'
+
+const props = defineProps<{
+  consoByHour: [string, number][]
+  prodByHour: [string, number][]
+}>()
+
+const productionData = convertDataForGraph(props.prodByHour)
+const consommationData = convertDataForGraph(props.consoByHour)
+const intersectionData = calculateIntersectionData(
+  productionData,
+  consommationData
+)
+const xAxis = generateXAxis()
+const tickPositions = generateTickPositions(xAxis)
+
+const chartOptions = computed(() => {
+  return {
+    chart: {
+      type: 'areaspline',
+      width: 730, // set the width of the chart
+      height: 400,
+      marginLeft: 30,
+    },
+    title: {
+      text: 'Production et consommation journalière',
+    },
+    legend: {
+      enabled: false,
+    },
+    xAxis: {
+      categories: xAxis,
+      labels: {
+        useHTML: true,
+        // @ts-ignore
+        formatter: function () {
+          // @ts-ignore
+          if (this.value === '') return ''
+          // @ts-ignore
+          return this.value === '12h' || this.value === '0h'
+            ? `<div style="width: 2px; height: 15px; background-color: black;" />`
+            : // @ts-ignore
+              `<span style="font-size: 9px;">${this.value}</span>`
+        },
+      },
+      rotation: 0,
+      tickPositions: tickPositions,
+      tickInterval: 1,
+    },
+    yAxis: {
+      title: {
+        text: 'Energie (kWh)',
+        align: 'high',
+        rotation: 0,
+        offset: 0,
+        y: -20,
+        x: 70,
+        style: {
+          fontSize: '11px',
+          fontWeight: 'bold',
+        },
+      },
+      labels: {
+        enabled: false,
+      },
+      lineWidth: 1,
+      gridLineWidth: 0,
+    },
+    tooltip: {
+      enabled: false,
+    },
+    plotOptions: {
+      areaspline: {
+        marker: {
+          enabled: false, // Disable data points
+        },
+        states: {
+          inactive: {
+            opacity: 1,
+          },
+          hover: {
+            enabled: false,
+          },
+        },
+      },
+    },
+    series: [
+      {
+        name: 'Consommation',
+        data: consommationData,
+        color: '#D1FAE5',
+        lineColor: '#0F766E',
+        fillOpacity: 0.6,
+      },
+      {
+        name: 'Production',
+        lineColor: '#F59E0B',
+        color: '#4F46E5',
+        data: productionData,
+        fillOpacity: 0.7,
+        lineWidth: 4,
+      },
+      {
+        name: 'Intersection',
+        data: intersectionData,
+        color: '#10B981',
+        lineWidth: 0,
+        fillOpacity: 1,
+      },
+    ],
+    credits: {
+      enabled: false, // Disable the Highcharts.com credit link
```

Comment:
do we need to buy the license? Or is there a free tier for our use case?

Reply:
According to the highchart website (https://www.highcharts.com/blog/download/) we can use highchart for free if Rennes is : "a non-profit organization, personal website, or school project ". But we have to display the credit 

![image](https://user-images.githubusercontent.com/121813008/235633445-f1e2b582-0200-45f8-84a4-81a1e58f7543.png)


I have change this line to display the credit